ams/ResetCommand.cs

```diff
@@ -32,67 +32,87 @@ internal class ResetCommand : BaseMigrator
 
         public override async Task MigrateAsync(CancellationToken cancellationToken)
         {
-            MediaServicesAccountResource? account = null;
-            try
+            var (isAMSAcc, account) = await IsAMSAccountAsync(_options.AccountName, cancellationToken);
+            if (!isAMSAcc)
             {
-                account = await GetMediaAccountAsync(_options.AccountName, cancellationToken);
+                var (storageClient, accountId) = await _resourceProvider.GetStorageAccount(_options.AccountName, cancellationToken);
+
+                _logger.LogInformation("Begin reset storage container: {name}", storageClient.AccountName);
+                double totalItems = await GetStorageBlobMetricAsync(accountId, cancellationToken);
+                var containers = storageClient.GetBlobContainersAsync(
+                               cancellationToken: cancellationToken);
+                _logger.LogInformation("The total containers count of the storage account is {count}.", totalItems);
+                List<BlobContainerItem>? filteredList = await containers.ToListAsync();
+                int resetAssetCount = 0;
+                foreach (var container in filteredList)
+                {
+                    BlobContainerClient containerClient = storageClient.GetBlobContainerClient(container.Name);
+                    await MigrateTaskAsync(containerClient, resetAssetCount, cancellationToken);
+                }
             }
-            catch (Exception)
+            else if (account != null)
```

Comment:
If IsAMSAccountAsync return true,  would the account is always not null, this check seems not right.

The similar code pattern applies to all other classes.